setup_resources.sh

```diff
@@ -0,0 +1,157 @@
+#!/bin/bash
+
+set -e
+
+# Function to validate RP running
+validate_rp_running() {
+    echo "########## Checking ARO RP Status ##########"
+    ELAPSED=0
+    while true; do
+        sleep 5
+        http_code=$(curl -k -s -o /dev/null -w '%{http_code}' https://localhost:8443/healthz/ready || true)
+        case $http_code in
+        "200")
+            echo "########## ✅ ARO RP Running ##########"
+            break
+            ;;
+        *)
+            echo "Attempt $ELAPSED - local RP is NOT up. Code : $http_code, waiting"
+            sleep 2
+            # after 40 secs return exit 1 to not block ci
+            ELAPSED=$((ELAPSED + 1))
+            if [ $ELAPSED -eq 20 ]; then
+                exit 1
+            fi
+            ;;
+        esac
+    done
+}
+
+# Ensure all env vars are set (CLUSTER_LOCATION, RESOURCEGROUP, CLUSTER_NAME)
+ALL_SET="true"
+if [ -z ${AZURE_SUBSCRIPTION_ID} ]; then ALL_SET="false" && echo "AZURE_SUBSCRIPTION_ID is unset"; else echo "AZURE_SUBSCRIPTION_ID is set to '$AZURE_SUBSCRIPTION_ID'"; fi
+if [ -z ${LOCATION} ]; then ALL_SET="false" && echo "LOCATION is unset"; else echo "LOCATION is set to '$LOCATION'"; fi
+if [ -z ${RESOURCEGROUP} ]; then ALL_SET="false" && echo "RESOURCEGROUP is unset"; else echo "RESOURCEGROUP is set to '$RESOURCEGROUP'"; fi
```

Comment:
I'd suggest changing this env var to a new one, e.g. `${CLUSTER_RESOURCEGROUP}`, to avoid any conflicts with reusing our existing `${RESOURCEGROUP}` env var used to hold the global dev env resources.

Reply:
1. Rename `RESOURCEGROUP` to `CLUSTER_RESOURCEGROUP`.
2. Update all references to `RESOURCEGROUP` accordingly.